pkg/cloudprovider/external.go

```diff
@@ -48,18 +52,53 @@ func isExternalFeatureGateEnabled(featureGate *configv1.FeatureGate) (bool, erro
 		// If no featureGate is present, then the user hasn't opted in to the external cloud controllers
 		return false, nil
 	}
+
+	enabledFeatureGates, disabledFeatureGates, err := getEnabledDisabledFeatureGates(featureGate)
+	if err != nil {
+		return false, err
+	}
+
+	return !disabledFeatureGates.Has(ExternalCloudProviderFeature) && enabledFeatureGates.Has(ExternalCloudProviderFeature), nil
+}
+
+// isCSIMigrationvSphereDisabled determines whether the CSIMigrationvSphere feature gate is disabled in the current
+// feature set. This function only returns true when the feature is actively disabled, otherwise we should
+// deploy the CCM for vSphere.
+// This function is needed to help address a reversion issue that is happening in 1.26 Kubernetes,
+// which is also being brought to 4.13 OpenShift. This feature gate had been locked for 1.26, but has been
+// unlocked in https://github.com/kubernetes/kubernetes/pull/116342. This feature gate had been removed from OpenShift
+// as all vSphere clusters will be migrated to the CSI driver, due to the reversion we must support a situation where
+// a user has upgraded to 4.13 but must revert their storage driver to in-tree. This reversion will also require the
+// CCM to be migrated back to the in-tree KCM. Related storage JIRA, https://issues.redhat.com/browse/STOR-1265
+// TODO remove this function once it is no longer needed, presumably 4.14 OpenShift.
+func isCSIMigrationvSphereDisabled(featureGate *configv1.FeatureGate) (bool, error) {
+	if featureGate == nil {
+		// If no featureGate is present, then the user hasn't opted in to the in-tree vSphere driver
+		return false, nil
+	}
+
+	enabledFeatureGates, disabledFeatureGates, err := getEnabledDisabledFeatureGates(featureGate)
+	if err != nil {
+		return false, err
+	}
+
+	return !disabledFeatureGates.Has(configv1.InTreeVSphereVolumes) && enabledFeatureGates.Has(configv1.InTreeVSphereVolumes), nil
```

Comment:
`InTreeVSphereVolumes` is the FeatureSet that disables the `CSIMigrationvSphere` FeatureGate [here](https://github.com/openshift/api/blob/ead9036371d0f77aff3dc6f3cdbaba081a774626/config/v1/types_feature.go#L132), so this line should check for `CSIMigrationvSphere` instead.

Reply:
ack, thanks!

Reply:
so, just to be clear here, the `CSIMigrationvSphere` needs to be in the disabled list?